src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReadThreadingAssemblerArgumentCollection.java

```diff
@@ -18,6 +18,7 @@
     private static final long serialVersionUID = 1L;
 
     public static final double DEFAULT_PRUNING_LOG_ODDS_THRESHOLD = MathUtils.log10ToLog(1.0);
+    public static final double DEFAULT_PRUNING_SEEDING_LOG_ODDS_THRESHOLD = MathUtils.log10ToLog(2.0);
```

Comment:
I would advocate this threshold be made very strict and that we do some reasoning about how probable it is for this to come up. It should be fairly unlikely for the seeding vertexes to ever fall off of good variation by accident due to low coverage I think.

Reply:
You're right, and I'll reason it out right now.  I don't know what conclusions this exercise will lead to. . .

The seeding method is fooled when an error chain A splits into two chains B and C, where B is the continuation of the original error and C is an additional error on top of the first, such that the likelihood of outgoing coverage N_C given incoming coverage N_A exceeds the threshold.

So let's first tabulate some values of N_A and N_C along with their log odds (from `Mutect2Engine.logLikelihoodRatio(N_B, N_C, errorRate)` as in the pruning code).  We'll use an error rate of 0.001.  Note that the current threshold is 4.605.

N_A  N_C  LOD
30     2      4.3
30     3      8.9
15     2      6.4
15     3      11.8
10     1       2.2
10     2       7.6
10     3       13.5
5       1       3.5
5       2       9.7
3       1        4.4
2        1        5.5

Without further calculation this already looks like the current threshold is too lenient and that raising the threshold significantly would not cause many real variants to fail.  Furthermore, even if a real variant with say 15 reads coming in and 2 coming out doesn't get counted as a seeding chain, it is overwhelmingly likely that the chain from which that 15-read chain is outgoing will be a seeding chain.  Thus I think this threshold can be raised to at least 10 with no harm to sensitivity whatsoever.  [To be continued. . .]

Reply:
So now let's think about false positive rates — that is, the probability of an error chain being used for seeding.  Note also that this isn't the end of the world.  It is not true that an error chain used for seeding starts a chain reaction of every error chain remaining unpruned, since chains still must pass the non-seeding log odds threshold.  Nonetheless, it's undesirable.

Suppose we have a coverage of N at some correct chain representing a real haplotype.  To get incorrect seeding we must have the same two errors sharing a kmer.  Otherwise the first error's chain will have returned to the non-error chain before the second error's chain begins.

If we have a seeding threshold of 10, the table above shows that we need a coincidence of errors somewhere in the ballpark of 5 for the first and two for the second.  At a coverage of 100 and an error rate of 0.001, the chance of this is 1 in ten million per locus.  The chance of two or more errors at any given locus is about 5 in 1000, so the chance of the second error site occurring within k = 25 bases is about one in ten.  If we have a 500-base window we would then expect a false seeding event one out of every 200 thousand assemblies.  Add to that the fairly small chance that the false seeding actually does any harm and I think we can safely say a seeding log odds threshold of 10 (this is all in log space, by the way) incurs no risk of under-pruning.